sql/opt_hints.h

```diff
@@ -516,6 +501,34 @@ class Opt_hints_qb : public Opt_hints
   const Parser::Subquery_hint *subquery_hint= nullptr;
   uint subquery_strategy= SUBS_NOT_TRANSFORMED;
 
+  /**
+    Returns TRUE if the query block has at least one of the hints
+    JOIN_PREFIX(), JOIN_SUFFIX(), JOIN_ORDER()  (but not JOIN_FIXED_ORDER()!)
+  */
+  bool has_join_order_hints() const
+  {
+    return join_order_hints.size() > 0;
+  }
+
+  /*
+    Adds a join order hint to the array if the capacity is not exceeded.
+    Returns FALSE on success,
+            TRUE on failure (capacity exceeded)
+  */
+  bool add_join_order_hint(Parser::Join_order_hint *hint_arg)
+  {
+    if (join_order_hints.size() >= MAX_ALLOWED_JOIN_ORDER_HINTS)
```

Comment:
Is there a situation where one could use more than one hint of any type?
I mean, there are four kinds of JOIN_XXX hints, any attempt to specify more than one hint of each kind causes an error:

```
MariaDB [test]> explain extended select /*+ JOIN_PREFIX(t2, t1) JOIN_PREFIX(t2)  */ * from t1 left join t2 on t2.a=t1.a where t2.b is not null;
...
| Warning | 4207 | Hint JOIN_PREFIX(`t2`) is ignored as conflicting/duplicated                                                                                                                                                                                            
```

So I was wondering what was the purpose of Opt_hints_qb::MAX_ALLOWED_JOIN_ORDER_HINTS=64  ?

Reply:
... is it because other SELECTs may specify hints for this SELECT ?

Reply:
No, multiple `JOIN_ORDER()` hints are allowed for a single query block, they are all applied. But `JOIN_PREFIX`, `JOIN_SUFFIX` and `JOIN_FIXED_ORDER` may be only one